src/charm.py

```diff
@@ -203,6 +203,23 @@ def _on_collect_status(self, e: CollectStatusEvent):
                     "metrics-generator disabled. Add a relation over send-remote-write"
                 )
             )
+        if (
+            # we have tls
+            self.coordinator.tls_available
+            # we have tls
+            and self.ingress.is_ready()
+            and self.ingress.scheme == "http"
```

Comment:
That's what we called "reverse termination".
- Strictly speaking it's not a bug
- I'd put this check on traefik side (lb server is https but traefik itself isn't)

Reply:
that makes sense

Reply:
opened an issue in traefik to track, closing this one